components/gfx/platform/macos/font.rs

```diff
@@ -61,8 +64,105 @@ impl FontTableMethods for FontTable {
 pub struct FontHandle {
     font_data: Arc<FontTemplateData>,
     ctfont: CTFont,
+    h_kern_subtable: Option<CachedKernTable>,
+    can_do_fast_shaping: bool,
+}
+
+impl FontHandle {
+    /// Cache all the data needed for basic horizontal kerning. This is used only as a fallback or
+    /// fast path (when the GPOS table is missing or unnecessary) so it needn't handle every case.
+    fn find_h_kern_subtable(&self) -> Option<CachedKernTable> {
+        let font_table = match self.table_for_tag(KERN) {
+            Some(table) => table,
+            None => return None
+        };
+
+        let mut result = CachedKernTable {
+            font_table: font_table,
+            pair_data_range: 0..0,
+            pixels_per_font_unit: 0.0,
+        };
+
+        // Look for a subtable with horizontal kerning in format 0.
+        // https://www.microsoft.com/typography/otspec/kern.htm
+        const KERN_COVERAGE_HORIZONTAL_FORMAT_0: u16 = 1;
+        const SUBTABLE_HEADER_LEN: usize = 6;
+        const FORMAT_0_HEADER_LEN: usize = 8;
+        {
+            let table = result.font_table.buffer();
+            let version = BigEndian::read_u16(table);
+            if version != 0 {
+                return None;
+            }
+            let num_subtables = BigEndian::read_u16(&table[2..]);
+            let mut start = 4;
+            for _ in 0..num_subtables {
```

Comment:
Kinda weird that the spec doesn't say what the subtable version number is supposed to be. Maybe add a TODO to maybe check this in the future?